eth/stagedsync/stage_headers.go

```diff
@@ -183,6 +184,24 @@ func HeadersPOW(
 Loop:
 	for !stopped {
 
+		// Only relevant to bor chain, called in case of milestone mismatch
```

Comment:
there's also some bor-specific code related to `borPenalties`, could we group the 2 somehow together?
ideally we should refactor stage_headers to be composable for a default case and bor case